### ISSUE TITLE
why is data not being formatted in multiple rows &  data ordering

### ISSUE THREAD

**Issue by ghost** [cimasim89/flat-file-creator#293]
I am setting my structure for writer and reader and the format of the test data I have is being generated **inline**. I was expecting to have multiple rows for each item formatted by the `rowDef`. Please help

**Comment by cimasim89** [comment]
Hi @joserenev , according to the interface of `getAsyncFlatFileCreator`, you can specify the rowEnd character on the option parameter, as you can see in  documentation the default beahviour  is to not specify a return character, adding the option `rowEnd:'\n'` you can make it works as you need.

**Comment by ghost** [comment]
@cimasim89 thanks for the clarification that solved my problem. I assumed that the library took care of breaking lines per data item. 
Another quick issue I am noting is that the data ordering in the output does not match the one passed to the function. Any idea reason why?

**Comment by cimasim89** [comment]
> @cimasim89 thanks for the clarification that solved my problem. I assumed that the library took care of breaking lines per data item. 

This is something we choose in our first release and change it now will be a breaking change



> Another quick issue I am noting is that the data ordering in the output does not match the one passed to the function. Any idea reason why?

Are you talking about the writer or the reader?

**Comment by ghost** [comment]
both do not match the original order of the data. It seems to be arbitrary changing order if I subsequently trigger writer and reader. I am expecting consistency in the output.

**Comment by cimasim89** [comment]
The order of the field in the positionary file is not given by the data object but by the maps array you specify before, the resulting position is the position of the map object in that array related by the `name` to your data object attribute, you can't use the data object because JavaScript object not preserve the position of object attributes.

**Comment by ghost** [comment]
Yes, the ordering of the fields for each row item is correct as specified in the `maps` array.  However, the order of the items
themself seem to be shifting apart from the original data.

Ex:

```
const rows: Array<MyData> = [
  {
    firstName: "Jo",
    lastName: "Revelo",
    dob: new Date("1986-01-01")
  },
  {
    firstName: "Lux",
    lastName: "Springfield",
    dob: new Date("1996-01-01")
  },
]

output file occasionally shifts items order to:

Lux   Springfield   1996-01-01
Jo    Revelo        1986-01-01


vs expected (given by the original data): 
Jo    Revelo        1986-01-01
Lux   Springfield   1996-01-01
```

**Comment by cimasim89** [comment]
let me see the map array `const rowDef: Array<FieldSpec> = [...]` that you are passing to `const createFile = getAsyncFlatFileCreator(rowDef)`

**Comment by ghost** [comment]
for testing purposes I am doing this and in the prev data example I removed `dob`:
```
        // Create row definition
        const rowDef: Array<FieldSpec> = [
          {
            name: "firstName",
            size: 8,
            type: "string",
          },
          {
            name: "lastName",
            size: 8,
            type: "string",
          },
        ];
```

**Comment by cimasim89** [comment]
ok i will try to investigate more about it

**Comment by ghost** [comment]
any update? @cimasim89

**Comment by ghost** [comment]
Also can we define structures with multiple row definitions in same file? Meaning that `rowDef[i] != rowDef[j] where i != j`. How do we handle this? @cimasim89

**Comment by ghost** [comment]
In `fileAppendPromise`  the method uses `fs.appendFile` which is called asynchronously and brings race condition to the file resource. Can we consider using the synchronous version `fs.appendFileSync` to maintain consistency in output?
please refer to: https://stackoverflow.com/a/57810170
@cimasim89
